delfin/drivers/dell_emc/vmax/client.py

```diff
@@ -51,19 +51,31 @@ def init_connection(self, access_info):
             LOG.error(msg)
             raise exception.StorageBackendException(msg)
 
+        if not self.uni_version:
+            msg = "Invalid input. Failed to get vmax unisphere version"
+            raise exception.InvalidInput(msg)
+
         self.array_id = access_info.get('extra_attributes', {}). \
             get('array_id', None)
 
-        # Get array details from unisphere
-        array = self.rest.get_array_detail(version=self.uni_version)
-        if len(array['symmetrixId']) == EMBEDDED_UNISPHERE_ARRAY_COUNT:
-            if not self.array_id:
-                self.array_id = array['symmetrixId'][0]
-            elif self.array_id != array['symmetrixId'][0]:
-                msg = "Invalid array_id. Supported id: {}". \
-                    format(array['symmetrixId'])
+        try:
+            # Get array details from unisphere
+            array = self.rest.get_array_detail(version=self.uni_version)
+            if not array:
+                msg = "Invalid input. Failed to connect to array"
                 raise exception.InvalidInput(msg)
```

Comment:
this could be because of n/w issues also.. Can we get all VMAX rest client exception and map this to delfin error codes? if matching error codes are not there , we can create..

Reply:
Changed error message, detailed exception will be handled as part of issue #184